### PR TITLE
go: add 1.21

### PR DESCRIPTION
**Pull request by imjasonh**
- [x] **IMPORTANT: 'image-request' tag has been applied if this PR is adding any images, including new versions or variants**

#### Quality Requirements for Images PRs
The items in this checklist should all be checked in the PR with exceptions clearly documented.
The general idea is that to the extent possible, the image should be a drop-in replacement to its public counterpart.
- [x] Image is a smaller size than its common public counterpart, or if not, the reason why should be explained
- [x] Image should be scanned for CVEs and should return 0 or near-0 results. Exceptions should be explained.
- [x] The software in the image should be able to be installed and work as expected
  - [x] The image should start in a cluster
  - [x] The application should be accessible to the user/cluster/etc. as expected
- [x] If there is an open source helm chart provided, the image should be able to be used with the chart. If needed, please add a -compat package to close any gaps with the public helm chart.
- [x] build for x86_64 and aarch64 if possible, document exceptions

#### For new image PRs only

If you have an apko.yaml file in this PR you need to follow this checklist, otherwise feel free to remove.
- [x] Include tests, sufficient enough that you would trust this image running in production.
- [x] Perform manual tests as needed and document your testing results
  - [x] For builder images (go, python, etc), build a sample app successfully
  - [ ] For services images (rabbit, databases, webservers) test basic functionality, upstream install/getting started, port availability, admin access. Document differences from public image.
  - [ ] Applications intended for use with a specific cloud provider must be tested on that cloud provider.

- [x] The version included is the latest GA version of the software
- [x] The latest tag points to the newest stable version
- [x] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [x] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
  - [x] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [x] The image contains the ca-certs bundle if needed
- [x] ENTRYPOINT
  - [x] For applications/servers/utilities call main program with no arguments e.g. [redis-server]
  - [x] For base images leave empty
  - [x] For dev variants set to entrypoint script that falls back to system
- [x] CMD:
  - [x] For server applications give arguments to start in daemon mode (may be empty)
  - [x] For utilities/tooling bring up help e.g. `–help`
  - [x] For base images with a shell, call it e.g. [/bin/sh]
- [x] Add annotations e.g:
```
annotations:
  "org.opencontainers.image.authors": "Chainguard Team https://www.chainguard.dev/"
  "org.opencontainers.image.url": https://edu.chainguard.dev/chainguard/chainguard-images/reference/busybox/ # use the academy site here
  "org.opencontainers.image.source": https://github.com/chainguard-images/images/tree/main/images/bazel # use github here
```
- [x] Check if environment variables are needed e.g. to set data locations
- [x] Ensure the image responds to SIGTERM
  - [x] `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`
- [x] Error logs write to stderr and normal logs to stdout. DO NOT write to file.
#### Documentation Requirements

- [x] Provide a README file, follow the [README template](readme-template.md) to include variants, tags, and usage examples
- [x] Document your testing and results in the PR
- [x] Document any exceptions to the testing and config requirements outlined in this template
- [x] Document where and how the image deviates from popular alternatives (ports, packages, user, entrypoint, versions, etc). Explain why there's a difference and if the user needs to do anything differently when using the image

### CODE REVIEW

File: images/go/configs/1.21.apko.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+contents:
+  keyring:
+    - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
+    - /Users/jason/git/wolfi-os/local-melange.rsa.pub
+  repositories:
+    - https://packages.wolfi.dev/os
+    - /Users/jason/git/wolfi-os/packages/
```
Comment (luhring) [inline images/go/configs/1.21.apko.yaml:7]: We can get rid of the `/Users...` lines

Comment (imjasonh) [reply images/go/configs/1.21.apko.yaml:7]: Nahhh 🙃

**Comment by luhring** [comment]
Does this need tests?

**Comment by imjasonh** [comment]
> Does this need tests?

It'll get tested by this: https://github.com/chainguard-images/images/blob/main/images/go/tests/01-version.sh

That's not a super thorough test, but it at least ensures the image contains go, can run, etc.

**Review by luhring — APPROVED** [review]
gopher

**Pull request merged**